src/frontend/src/optimizer/plan_node/logical_row_id_gen.rs

```diff
@@ -0,0 +1,122 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{fmt, vec};
+
+use risingwave_common::bail;
+use risingwave_common::error::Result;
+
+use super::{
+    ColPrunable, ExprRewritable, LogicalFilter, PlanBase, PlanRef, PlanTreeNodeUnary,
+    PredicatePushdown, StreamRowIdGen, ToBatch, ToStream,
+};
+use crate::optimizer::plan_node::{
+    ColumnPruningContext, PredicatePushdownContext, RewriteStreamContext, ToStreamContext,
+};
+use crate::optimizer::property::FunctionalDependencySet;
+use crate::utils::{ColIndexMapping, Condition};
+
+/// `LogicalRowIdGen` generates `_row_id` at `row_id_index`
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct LogicalRowIdGen {
```

Comment:
IIUC, we do not leverage `RowIdGen` for `StreamValues`, right? 🤔 So why do we include this change in this PR? 👀